core/src/test/java/io/parsingdata/metal/expression/value/FoldCatTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.value;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import static io.parsingdata.metal.Shorthand.cat;
+import static io.parsingdata.metal.Shorthand.con;
+import static io.parsingdata.metal.Shorthand.div;
+import static io.parsingdata.metal.Shorthand.ref;
+import static io.parsingdata.metal.Shorthand.seq;
+import static io.parsingdata.metal.expression.value.BytesTest.EMPTY_PARSE_STATE;
+import static io.parsingdata.metal.util.EncodingFactory.enc;
+import static io.parsingdata.metal.util.ParseStateFactory.stream;
+import static io.parsingdata.metal.util.TokenDefinitions.any;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import org.junit.Test;
+
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.data.ImmutableList;
+import io.parsingdata.metal.data.ParseState;
+
+public class FoldCatTest {
+
+    public static final Optional<ParseState> RESULT =
+        seq(any("a"),
```

Comment:
At first I was a bit confused about "abc" and 3 any's with an "a".
If you know that the "a" in any is the name, that it becomes more clear what happens here.

Reply:
Never thought about it this way because that string being used as name is so natural to me. But I totally agree it's confusing. I've changed the name to "any" to better signal it has nothing to do with the value.